### ISSUE TITLE
IOU, GIOU, Class,Obj are 0.000000

### ISSUE THREAD

**Issue by markgao-916** [AlexeyAB/darknet#5847]
OpenCV exception: wait_key_cv 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.535068, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 2070.212891, iou_loss = 0.000000, total_loss = 2070.212891 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.495549, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 455.184631, iou_loss = 0.000000, total_loss = 455.184631 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.495070, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 111.342873, iou_loss = 0.000000, total_loss = 111.342873 
OpenCV exception: wait_key_cv 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.534216, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 2066.980957, iou_loss = 0.000000, total_loss = 2066.980957 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.494164, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 453.285828, iou_loss = 0.000000, total_loss = 453.285828 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.493949, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 110.886040, iou_loss = 0.000000, total_loss = 110.886040 
OpenCV exception: wait_key_cv 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.533769, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 2057.534424, iou_loss = 0.000000, total_loss = 2057.534424 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.495600, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 454.045776, iou_loss = 0.000000, total_loss = 454.045776 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.496387, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 111.524506, iou_loss = 0.000000, total_loss = 111.524506 
OpenCV exception: wait_key_cv 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.534931, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 2066.900879, iou_loss = 0.000000, total_loss = 2066.900879 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.495223, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 455.410187, iou_loss = 0.000000, total_loss = 455.410187 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.495417, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 111.875847, iou_loss = 0.000000, total_loss = 111.875847 

This is the logs printed in screen, i do not know the reason why too many 0.00000, my task is to detect car and bus, my classes=2, filters=21

**Comment by WongKinYiu** [comment]
could you provide your cfg file and results of `darknet.exe detector calc_anchors data/obj.data -num_of_clusters 9 -width 416 -height 416`

**Comment by markgao-916** [comment]
this is my anchors
anchors = 12, 11, 16, 16, 21, 16, 23, 21, 25, 27, 37, 32, 48, 45, 70, 61, 108, 100
which is calculated by k-means, 
and i get the error while calculate anchors by darknet
![图片](https://user-images.githubusercontent.com/50621508/83875128-26a56680-a769-11ea-9d75-d96c4dd5a101.png)

yolo layer setting in my cfg as follow:

[yolo]
mask = 0,1,2
#anchors = 12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401
anchors = 12, 11, 16, 16, 21, 16, 23, 21, 25, 27, 37, 32, 48, 45, 70, 61, 108, 100
classes=2
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
scale_x_y = 1.2
iou_thresh=0.213
cls_normalizer=1.0
iou_normalizer=0.07
iou_loss=ciou
nms_kind=greedynms
beta_nms=0.6
max_delta=5


[route]
layers = -4

[convolutional]
batch_normalize=1
size=3
stride=2
pad=1
filters=256
activation=leaky

[route]
layers = -1, -16

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=512
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=512
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=512
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=21
activation=linear


[yolo]
mask = 3,4,5
#anchors = 12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401
anchors = 12, 11, 16, 16, 21, 16, 23, 21, 25, 27, 37, 32, 48, 45, 70, 61, 108, 100
classes=2
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
scale_x_y = 1.1
iou_thresh=0.213
cls_normalizer=1.0
iou_normalizer=0.07
iou_loss=ciou
nms_kind=greedynms
beta_nms=0.6
max_delta=5


[route]
layers = -4

[convolutional]
batch_normalize=1
size=3
stride=2
pad=1
filters=512
activation=leaky

[route]
layers = -1, -37

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=21
activation=linear


[yolo]
mask = 6,7,8
#anchors = 12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401
anchors = 12, 11, 16, 16, 21, 16, 23, 21, 25, 27, 37, 32, 48, 45, 70, 61, 108, 100
classes=2
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=0
scale_x_y = 1.05
iou_thresh=0.513
cls_normalizer=1.0
iou_normalizer=0.07
iou_loss=ciou
nms_kind=greedynms
beta_nms=0.6
max_delta=5

**Comment by WongKinYiu** [comment]
please use default anchors to train and then post your results.

**Comment by markgao-916** [comment]
I have trained the natwork use default anchors, but the result is the same
![图片](https://user-images.githubusercontent.com/50621508/83985990-cb909100-a96d-11ea-8969-4f3189c3aa20.png)

**Comment by AlexeyAB** [comment]
If you have an issue with training - no-detections / Nan avg-loss / low accuracy:
    * what command do you use?
    * what dataset do you use?   
    * what Loss and mAP did you get?
    * show chart.png with Loss and mAP    
    * check your dataset - run training with flag `-show_imgs` i.e. `./darknet detector train ... -show_imgs` and look at the `aug_...jpg` images, do you see correct truth bounded boxes?
    * rename your cfg-file to txt-file and drag-n-drop (attach) to your message here
    * show content of generated files `bad.list` and `bad_label.list` if they exist
    * Read `How to train (to detect your custom objects)` and `How to improve object detection` in the Readme: https://github.com/AlexeyAB/darknet/blob/master/README.md
    * show such screenshot with info
```
./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights data/dog.jpg
 CUDA-version: 10000 (10000), cuDNN: 7.4.2, CUDNN_HALF=1, GPU count: 1
 CUDNN_HALF=1
 OpenCV version: 4.2.0
 0 : compute_capability = 750, cudnn_half = 1, GPU: GeForce RTX 2070
net.optimized_memory = 0
mini_batch = 1, batch = 8, time_steps = 1, train = 0
   layer   filters  size/strd(dil)      input                output
```

**Comment by markgao-916** [comment]
The formate of annotations is wrong, Thank you very much @AlexeyAB

**Comment by KelvinHuang666** [comment]
I can train yolov4 with my own dataset,but cant train yolov3
ie:(IOU: 0.000000, GIOU: 0.000000), Class: 0.000000, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, I use yolov3.cfg with change classes and filter
how can I solve it,hhanks
